src/markup.ts

```diff
@@ -0,0 +1,394 @@
+import {
+  ForceReply,
+  InlineKeyboardButton,
+  InlineKeyboardMarkup,
+  KeyboardButton,
+  MessageEntity,
+  ReplyKeyboardMarkup,
+  ReplyKeyboardRemove,
+} from 'typegram'
+import { is2D } from './util'
+
+// functions for HTML tag escaping based on https://stackoverflow.com/a/5499821/
+const tagsToEscape = {
+  '&': '&amp;',
+  '<': '&lt;',
+  '>': '&gt;',
+}
+function escapeTag(tag: string): string {
+  return tagsToEscape[tag as '&' | '<' | '>'] ?? tag
+}
+function escapeHTML(str: string): string {
+  return str.replace(/[&<>]/g, escapeTag)
+}
+
+type MarkupBuilder = Partial<
+  InlineKeyboardMarkup & ReplyKeyboardMarkup & ReplyKeyboardRemove & ForceReply
+>
+
+type Hideable<B> = B & { hide: boolean }
+type HideableKBtn = Hideable<KeyboardButton>
+type HideableIKBtn = Hideable<InlineKeyboardButton>
+
+class Markup {
+  public readonly reply_markup: MarkupBuilder = {}
```

Comment:
Ideally, this should be similar to `InlineKeyboardMarkup | ReplyKeyboardMarkup | ReplyKeyboardRemove | ForceReply`... I'll try to make it happen when I'll have time to spare.

Reply:
I'd make `Markup` generic and immutable to achieve this...